Library/Homebrew/cmd/update.sh

```diff
@@ -9,6 +9,14 @@
 # shellcheck source=/dev/null
 source "$HOMEBREW_LIBRARY/Homebrew/utils/lock.sh"
 
+git() {
```

Comment:
It does. It should speed up the `git config` calls in `utils/analytics.sh` a bit. There are typically three of them and they happen synchronously during analytics setup, thus slow down every `brew` run (except for special cases like `--prefix`). All of this happens before the analytics are sent in a `curl` process in the background.